Patch extracted from Debian's shiboken package (1.2.2-3.1 for Debian "stretch")

Copyright: 2009-2016, Didier Raboud <odyx@debian.org>
License: LGPL-2.1+

Description: don't *2 the object's identifier because this causes an
 unsigned long overflow when the result is > ULONG_MAX
Author: Gilles Filippini <pini@debian.org>
Bug-Debian: http://bugs.debian.org/856133
Index: shiboken-1.2.2/tests/libother/otherobjecttype.cpp
===================================================================
--- shiboken-1.2.2.orig/tests/libother/otherobjecttype.cpp
+++ shiboken-1.2.2/tests/libother/otherobjecttype.cpp
@@ -25,6 +25,6 @@
 Collector&
 operator<<(Collector& collector, const OtherObjectType& obj)
 {
-    collector << obj.identifier()*2;
+    collector << obj.identifier();
     return collector;
 }
Index: shiboken-1.2.2/tests/otherbinding/collector_external_operator_test.py
===================================================================
--- shiboken-1.2.2.orig/tests/otherbinding/collector_external_operator_test.py
+++ shiboken-1.2.2/tests/otherbinding/collector_external_operator_test.py
@@ -46,7 +46,7 @@ class CollectorOtherObjectType(unittest.
         collector = Collector()
         obj = OtherObjectType()
         collector << obj
-        self.assertEqual(collector.items()[0], obj.identifier() * 2)
+        self.assertEqual(collector.items()[0], obj.identifier())
 
 if __name__ == '__main__':
     unittest.main()
